content/200-concepts/100-components/03-prisma-migrate/010-mental-model.mdx

```diff
@@ -0,0 +1,137 @@
+
+![Introspect your database with Prisma](https://res.cloudinary.com/prismaio/image/upload/v1628761155/docs/f7itiYw.png)
+
+- **Model/Entity-first migration:** with this pattern, you define the structure of the database schema with code and then use a migration tool to generate the SQL, for example, for syncing your application and database schema.
+
+![Evolve your database schema with Prisma Migrate](https://user-images.githubusercontent.com/33921841/189865730-f6f8b402-4f6e-4a4b-b168-ba615bf51ca2.png)
+
+Artifacts that describe schema updates should ideally be stored together with your application code. They should also be tracked in version control and shared with the rest of the team working on the application.
+
+Migrations provide _state management_ which helps you to track the state of the database.
+
+Migrations also allow you to replicate the state of a database at a specific point in time which is useful when collaborating with other members of the team, e.g. switching between different branches.
+
+For further information on database migrations, see the [Prisma Data Guide](https://www.prisma.io/dataguide/types/relational/what-are-database-migrations).
+
+## What is Prisma Migrate?
+
+Prisma Migrate is a database migration tool that supports _code first_ and _database first_ patterns to manage database schemas in your local environment and in production.
+
+The workflow when using Prisma Migrate in your project would be iterative and look like this:
+
+- Define your Prisma schema – the source of truth
+- Use one of the [Prisma CLI commands](/reference/api-reference/command-reference) to sync your Prisma schema with the database schema
+
+![Prisma Migrate workflow](https://user-images.githubusercontent.com/33921841/189882708-8d9e4e51-9449-4216-baf7-ac0eda0410a2.png)
+
+Prisma Migrate uses the following pieces of state to track the state of your database schema:
+
+- **Prisma schema**: the main configuration file. This contains your data source, generators, and data models.
+- **Migrations history**: the history of changes made to your database schema represented in the `prisma/migrations` folder.
+- **Migrations table**: the table in the database that stores metadata of the already applied migrations .(`_prisma_migrations`)
+- **Database schema**: the state of the database.
+
+![Prisma Migrate "state management"](https://user-images.githubusercontent.com/33921841/189885586-0e0d7852-8ac7-4842-a6cb-bd158c20d28e.png)
+
+## Evolve your database schema with Prisma Migrate
+
+### Assumptions when working with Prisma Migrate
+
+- Ideally, you should use different databases for different environments. For example, you might have a separate database for development, preview, and production environments.
+- The databases you use in development environments are disposable — you can easily create, use, and delete on demand.
+- The database configuration used in different environments should be similar to ensure it is reproducible.
```

Comment:
What does 'similar' mean?

Reply:
Using a PosgreSQL database with similar server configuration in development, staging, and production.

Reply:
I would spell this out